op-txproxy/auth_handler.go

```diff
@@ -56,18 +59,17 @@ func (h *authHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		r.Body = io.NopCloser(bytes.NewBuffer(nil))
 	}
 
-	// The request body can only be read once. Adjust here so it's inspectable
-	var buf bytes.Buffer
-	body := io.TeeReader(r.Body, &buf)
-	r.Body = io.NopCloser(&buf)
-
 	// Since this middleware runs prior to the server, we need to manually apply the body limit when reading.
-	bodyBytes, err := io.ReadAll(io.LimitReader(body, int64(defaultBodyLimit)))
+	bodyBytes, err := io.ReadAll(io.LimitReader(r.Body, int64(defaultBodyLimit)))
 	if err != nil {
 		http.Error(w, "unable to parse request body", http.StatusInternalServerError)
 		return
 	}
 
+	// The request body can only be read once so place back in the bytes along with whatever else may have been present
+	defer r.Body.Close()
+	r.Body = io.NopCloser(io.MultiReader(bytes.NewReader(bodyBytes), r.Body))
```

Comment:
Forward responsibility and the ability to close to the next handler.
```suggestion
    r.Body = interface{
        io.Reader
        io.Closer
    }{
        io.MultiReader(bytes.NewReader(bodyBytes), r.Body)),
        r.Body,
    }
```

Reply:
ah niiice this is sleek